### PR TITLE
docs - update transaction isolation level info

### PR DESCRIPTION
**Pull request by lisakowen**
received some feedback that the transaction isolation level documentation was confusing and inaccurate.  in this PR:
- update the topic to align more with the postgres doc content (https://www.postgresql.org/docs/12/transaction-iso.html).
- continue to call out greenplum-specifics (doesn't support SERIALIZABLE - retained the existing doc info, didn't include the postgres serializable topic).
- also added a topic about setting the isolation level at the {cluster, session, transaction} level.

doc review site link (behind vpn) - will need to scroll up a little:  https://docs-staging.vmware.com/en/draft/VMware-Tanzu-Greenplum/trans-iso-level/greenplum-database/GUID-admin_guide-intro-about_mvcc.html#transaction-isolation-levels

### CODE REVIEW

File: gpdb-doc/markdown/admin_guide/intro/about_mvcc.html.md [hunk 1]
```diff
@@ -56,54 +56,120 @@ See [Recovering from a Transaction ID Limit Error](../managing/maintain.html#np1
...
 
-## <a id="section_f4m_n5n_fs"></a>Transaction Isolation Modes 
+## <a id="section_f4m_n5n_fs"></a>Transaction Isolation Levels 
 
-The SQL standard describes three phenomena that can occur when database transactions run concurrently:
+The SQL standard defines four levels of transaction isolation. The most strict is Serializable, which is defined by the standard in a paragraph which says that any concurrent execution of a set of Serializable transactions is guaranteed to produce the same effect as running them one at a time in some order. The other three levels are defined in terms of phenomena, resulting from interaction between concurrent transactions, which must not occur at each level. The standard notes that due to the definition of Serializable, none of these phenomena are possible at that level.
```
Comment (dyozie) [inline gpdb-doc/markdown/admin_guide/intro/about_mvcc.html.md:61]: I realize this is probably taken from postgres docs.  But it really seems like the following text should be removed:  "which is defined by the standard in a paragraph"

I guess maybe it's referring to a paragraph in the SQL standard?  Even so, I find it confusing.

File: gpdb-doc/markdown/admin_guide/intro/about_mvcc.html.md [hunk 2]
```diff
@@ -56,54 +56,120 @@ See [Recovering from a Transaction ID Limit Error](../managing/maintain.html#np1
...
+|Isolation Level|Dirty Read|Non-Repeatable|Phantom Read|Serialization Anomoly|
+|-----|----------|--------------|------------|------|
+|`READ UNCOMMITTED`|Allowed, but not in Greenplum Database|Possible|Possible|Possible|
+|`READ COMMITTED`|Impossible|Possible|Possible|Possible|
+|`REPEATABLE READ`|Impossible|Impossible|Allowed, but not in Greenplum Database|Possible|
+|`SERIALIZABLE`|Impossible|Impossible|Impossible|Impossible|
 
-The difference between `READ COMMITTED` and `REPEATABLE READ` is that with `READ COMMITTED`, each statement in a transaction sees only rows committed before the *statement* started, while in `READ COMMITTED` mode, statements in a transaction see only rows committed before the *transaction* started.
+In Greenplum Database, you can request any of the four transaction isolation levels, but only two distinct isolation levels are implemented internally. The Greenplum Database `READ UNCOMMITTED` level behaves like `READ COMMITTED`, and the `SERIALIZABLE` level falls back to `REPEATABLE READ`.
```
Comment (dyozie) [inline gpdb-doc/markdown/admin_guide/intro/about_mvcc.html.md:79]: Maybe reword first sentence to:  Greenplum Database implements only two distinct isolation levels, although you can request any of the four described transaction isolation levels.

File: gpdb-doc/markdown/admin_guide/intro/about_mvcc.html.md [hunk 3]
```diff
@@ -56,54 +56,120 @@ See [Recovering from a Transaction ID Limit Error](../managing/maintain.html#np1
...
+### <a id="til_rc"></a>Read Committed Isolation Level 
 
-The SQL `SET TRANSACTION ISOLATION LEVEL` statement sets the isolation mode for the current transaction. The mode must be set before any `SELECT`, `INSERT`, `DELETE`, `UPDATE`, or `COPY` statements:
+The default isolation level in Greenplum Database is `READ COMMITTED`. When a transaction uses this isolation level, a `SELECT` query \(without a `FOR UPDATE/SHARE` clause\) sees only data committed before the query began; it never sees either uncommitted data or changes committed during query execution by concurrent transactions. In effect, a `SELECT` query sees a snapshot of the database as of the instant the query begins to run. However, `SELECT` does see the effects of previous updates executed within its own transaction, even though they are not yet committed. Also note that two successive `SELECT` commands can see different data, even though they are within a single transaction, if other transactions commit changes after the first `SELECT` starts and before the second `SELECT` starts.
```
Comment (dyozie) [inline gpdb-doc/markdown/admin_guide/intro/about_mvcc.html.md:90]: as of -> at

Comment (dyozie) [reply gpdb-doc/markdown/admin_guide/intro/about_mvcc.html.md:90]: it is the updated version of the row that is locked -> the the updated version of the row is locked

**Review by dyozie — APPROVED** [review]

**Comment by lisakowen** [comment]
thank you for reviewing, @dyozie !

**Review by SmartKeyerror — APPROVED** [review]
LGTM.
